File: ObserverDesignPattern/EventSubscription_Example/Driver.java
package ObserverDesignPattern.EventSubscription_Example;

public class Driver {
	public static void main(String[] args) {
		Editor editor = new Editor();
		editor.queue.subscribeEvents(new EmailEventListener("aadityakasbekar@gmail.com"),EventTypes.DELETE, EventTypes.UPDATE, EventTypes.RENAME);
		editor.queue.subscribeEvents(new LoggerEventListener("/logs.txt"), EventTypes.values());

		try {
			editor.openFile("/test.txt");
			editor.updateFile();
			editor.saveFile();
			editor.renameFile();
			editor.deleteFile();
			editor.closeFile();
//			editor.updateFile(); //just checking if the exception handling is done correctly or not.
		} catch (Exception e) {
			e.printStackTrace();
		}
	}
}

File: ObserverDesignPattern/EventSubscription_Example_Python/Driver.py
from Subjects.Editor import Editor
from Subjects.EventTypes import EventTypes
from Observers.EmailService import EmailService
from Observers.LoggerService import LoggerService

class Driver:

  @staticmethod
  def main():
    editor : Editor = Editor()
    editor.queue.registerObserver(EmailService("aadityakasbekar.com"), EventTypes.RENAME, EventTypes.DELETE, EventTypes.UPDATE)
    editor.queue.registerObserver(LoggerService("logs.txt"), *(event for event in EventTypes))

    editor.openFile("test.txt")
    editor.updateFile()
    # editor.saveFile()
    editor.renameFile("test.txt")
    editor.closeFile("test.txt")
    editor.deleteFile("test1.txt")

if __name__ == "__main__":
  Driver.main()